src/Microsoft.Performance.SDK.Runtime.NetCoreApp/Plugins/PluginsLoader.cs

```diff
@@ -0,0 +1,255 @@
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if all plugins within the given directory were loaded, <c>false</c> otherwise.
+        /// </returns>
+        /// <remarks>
+        ///     If multiple plugins are contained within the given directory, this method will return
+        ///     <c>false</c> if _any_ of them fail to load and not provide information as to which
+        ///     plugin(s) failed. For example, if the filesystem looks like
+        ///         rootFolder
+        ///             pluginA
+        ///                 ...files for pluginA...
+        ///             pluginB
+        ///                 ...files for pluginB...
+        ///     TryLoadPlugin(rootFolder) will fail if either or both of pluginA or pluginB fail to load.
+        ///     If the caller must know which plugin(s) failed, it must use <see cref="TryLoadPlugins(IEnumerable{string}, out IList{string})"/>.
+        ///     <para/>
+        ///     This method does not report an error if a different version of one of the custom data
+        ///     sources being loaded has already been loaded by either the current or a previous call to <see cref="TryLoadPlugin(string)"/>
+        ///     or <see cref="TryLoadPlugins(IEnumerable{string}, out IList{string})"/>.
+        /// </remarks>
+        public bool TryLoadPlugin(string directory)
+        {
+            return this.TryLoadPlugins(new List<string> { directory }, out _);
+        }
+
+        /// <summary>
+        ///     Attempts to load the plugin(s) contained within any of the given directories.
+        /// </summary>
+        /// <param name="directories">
+        ///     The directories to attempt to load plugins from.
+        /// </param>
+        /// <param name="failed">
+        ///     The directories from <paramref name="directories"/> which contained at least
+        ///     one plugin that failed to load. If this method returns <c>true</c>, this
+        ///     will be empty.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if all plugins within the given directories were loaded, <c>false</c> otherwise.
+        /// </returns>
+        /// <remarks>
+        ///     A directory from <paramref name="directories"/> will be included in <paramref name="failed"/>
+        ///     if _any_ of the plugins inside the directory fail to load. See remarks on <see cref="TryLoadPlugin(string)"/>
+        ///     for more information.
+        ///     <para/>
+        ///     This method does not report an error if a different version of one of the custom data
+        ///     sources being loaded has already been loaded by either the current or a previous call to <see cref="TryLoadPlugin(string)"/>
+        ///     or <see cref="TryLoadPlugins(IEnumerable{string}, out IList{string})"/>.
+        /// </remarks>
+        public bool TryLoadPlugins(IEnumerable<string> directories, out IList<string> failed)
```

Comment:
>TryLoadPlugins [](start = 20, length = 14)

does it make sense to add an Async version of these calls?

Reply:
I will add some async wrappers for convenience